macros/src/types/named.rs

```diff
@@ -92,22 +93,33 @@ fn format_field(
         return Ok(());
     }
 
+    if type_as.is_some() && type_override.is_some() {
+        syn_err!("`type` is not compatible with `as`")
+    }
+
+    let parsed_ty = if let Some(ref type_as) = type_as {
+        syn::parse_str::<Type>(type_as)?
+    } else {
+        field.ty.clone()
+    };
+
     let (ty, optional_annotation) = match optional {
         Optional { optional: true, nullable } => {
             let inner_type = extract_option_argument(&field.ty)?; // inner type of the optional
             match nullable {
-                true => (&field.ty, "?"),  // if it's nullable, we keep the original type
+                true => (&parsed_ty, "?"),  // if it's nullable, we keep the original type
                 false => (inner_type, "?"), // if not, we use the Option's inner type
```

Comment:
The thing I'm most confused on how to do is this. This code branch bypasses `type_as`

Reply:
Note: I'm talking about line 111, not the lines marked in the diff

Reply:
Just hit that issue as well. I'll have to think about how it *should* interact, i'm not 100% sure about that yet.